korgw/src/commonMain/kotlin/com/soywiz/korev/Events.kt

```diff
@@ -82,11 +90,100 @@ data class Touch(
         this.id = other.id
         this.x = other.x
         this.y = other.y
+        this.force = other.force
         this.status = other.status
+        this.kind = other.kind
     }
 
     override fun hashCode(): Int = index
     override fun equals(other: Any?): Boolean = other is Touch && this.index == other.index
+
+    fun toStringNice() = "Touch[${id}][${status}](${x.niceStr},${y.niceStr})"
+}
+
+// On JS: each event contains the active down touches (ontouchend simply don't include the touch that has been removed)
+// On Android: ...
+// On iOS: each event contains partial touches with things that have changed for that specific event
+class TouchBuilder {
```

Comment:
Is there any reason to have all the implementations in the common source set? For now, I guess implementations can be moved to platform source sets via expect/actual, so generated binaries won't contain redundant implementations.

Reply:
I didn't want to use `JS, Android, IOS` as part of the enum, but something like `PROVIDE_EVENT_TOUCHES`, `PROVIDE_ACTIVE_TOUCHES`, `PROVIDE_ALL_TOUCHES`, so it is more like a behaviour of which touches are provided in each event more than the platform itself, that could be reused for several platforms. But since I didn't come with good names yesterday, I named them like that lol 🤣. This API shouldn't be used by most people so we can change it.
Android is not using TouchBuilder IIRC, so maybe I can provide only the two implementations, or check how are android providing touches and using one of the two or add the other one. In the end I believe implementations could provide touches on each even in those three ways, and this tries to normalize that